### DISCUSSION TITLE
How do I manually add Main Branch into HA

### DISCUSSION THREAD

**Discussion by ghgeiger** [hassio-addons/app-tor#126] [Q&A] [answered]
Right now this add-on is useless to me without being able to generate a v3 .onion address. I have been patiently waiting for the main branch to come through as an update in the add-on store so that I can start using tor hidden services again on my system, but it's been over two months & I have no idea when or if that will actually happen. I have successfully manually added several other repositories into my system, but I can't figure out the correct url to add this add-on's main branch. I have tried to add:
https://github.com/hassio-addons/addon-tor
and
https://github.com/hassio-addons/addon-tor/tree/main/tor
but both of those urls generate the following error: "Invalid Add-on repository!" with a similar error in the supervisor logs.

Can someone point me in the right direction?
Thanks

**Comment by frenck** [comment, marked as answer]
https://developers.home-assistant.io/docs/add-ons

**Comment by ghgeiger** [comment]
Thanks for the response frenk,

I take it that the proper way to do this is to build my own docker container & publish the add on locally? Unfortunately that's not a viable option for me right now. Can you give us any kind of timeline or estimate of when 3.06 might be pushed out?

**Reply by frenck** [reply]
> I take it that the proper way to do this is to build my own docker container & publish the add on locally? Unfortunately that's not a viable option for me right now. 

Yes, that is what you asked?

>  Can you give us any kind of timeline or estimate of when 3.06 might be pushed out?

I don't work with timelines or agendas.
